app/src/main/java/org/wikipedia/util/DeviceUtil.kt

```diff
@@ -74,6 +76,28 @@ object DeviceUtil {
         }
     }
 
+    /**
+     * https://issuetracker.google.com/issues/160946170
+     * There's a platform-specific issue where the app gets launched in "Restricted" mode during a
+     * backup operation; And if the app crashes during that operation, it remains in Restricted
+     * mode in subsequent launches, including subsequent user-requested launches. While in this
+     * mode, the system doesn't actually launch our custom subclassed WikipediaApp object, but
+     * instead uses a vanilla Application object, which will cause issues when other classes try
+     * to access static data from the WikipediaApp object.
+     *
+     * This is a workaround that explicitly terminates the app process if it's running in Restricted
+     * mode, to be used sparingly from places where this crash is most likely to occur.
+     */
+    fun assertAppContext(context: Context, terminateOnFail: Boolean = false): Boolean {
+        if (context.applicationContext !is WikipediaApp) {
+            if (terminateOnFail) {
+                Handler(context.mainLooper).post { exitProcess(0) }
```

Comment:
Since we're using `post`, should we implement error handling with a try-catch around the Handler?

Reply:
@cooltey What Exceptions would we be catching?
It's true that we're very careful when using `post` in an Activity, making sure it's not destroyed, but this `post` is not attached to any activity, and if the `exitProcess()` function throws an exception (?), I'd like to see it in the crash logs.

Reply:
Didn't see any crashes in testing and it makes sense to me.